01_fib/fibonacci.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+	var (
+		fn   = 1
+		f0   int
+		f1   = 1
+		list []int
+	)
+
+	for i := 0; i < n; i++ {
+		if i > 0 {
+			fn = f0 + f1
+			f0, f1 = f1, fn
+		}
+		list = append(list, fn)
+	}
+	return list
+}
+
+func fib(n int) {
+	//ideally this can be printed in main but lab needs to be printed nside the func
+	fmt.Fprint(out, calculateFib(n))
```

Comment:
However, this does not print the requested new line separated list.
Use https://golang.org/pkg/strings/#Join to achieve this.